nifi-commons/nifi-security-utils/src/test/java/org/apache/nifi/security/util/crypto/Argon2CipherProviderTest.java

```diff
@@ -0,0 +1,418 @@
+        }
+    }
+
+    @Test
+    void testGetCipherWithExternalIVShouldBeInternallyConsistent() throws Exception {
+        // Arrange
+        final String PASSWORD = "shortPassword";
+        final byte[] SALT = cipherProvider.generateSalt();
+        final byte[] IV = Hex.decodeHex("01".repeat(16).toCharArray());
+
+        // Act
+        for (EncryptionMethod em : strongKDFEncryptionMethods) {
+            // Initialize a cipher for encryption
+            Cipher cipher = cipherProvider.getCipher(em, PASSWORD, SALT, IV, DEFAULT_KEY_LENGTH, true);
+
+            byte[] cipherBytes = cipher.doFinal(PLAINTEXT.getBytes("UTF-8"));
+
+            cipher = cipherProvider.getCipher(em, PASSWORD, SALT, IV, DEFAULT_KEY_LENGTH, false);
+            byte[] recoveredBytes = cipher.doFinal(cipherBytes);
+            String recovered = new String(recoveredBytes, "UTF-8");
+
+            // Assert
+            assertEquals(PLAINTEXT, recovered);
+        }
+    }
+
+    @Test
+    void testGetCipherWithUnlimitedStrengthShouldBeInternallyConsistent() throws Exception {
+        // Arrange
+        final String PASSWORD = "shortPassword";
+        final byte[] SALT = cipherProvider.generateSalt();
+
+        final int LONG_KEY_LENGTH = 256;
+
+        // Act
+        for (EncryptionMethod em : strongKDFEncryptionMethods) {
+            // Initialize a cipher for encryption
+            Cipher cipher = cipherProvider.getCipher(em, PASSWORD, SALT, LONG_KEY_LENGTH, true);
+            byte[] iv = cipher.getIV();
+
+            byte[] cipherBytes = cipher.doFinal(PLAINTEXT.getBytes("UTF-8"));
+
+            cipher = cipherProvider.getCipher(em, PASSWORD, SALT, iv, LONG_KEY_LENGTH, false);
+            byte[] recoveredBytes = cipher.doFinal(cipherBytes);
+            String recovered = new String(recoveredBytes, "UTF-8");
+
+            // Assert
+            assertEquals(PLAINTEXT, recovered);
+        }
+    }
+
+    @Test
+    void testGetCipherShouldNotAcceptInvalidSalts() throws Exception {
+        // Arrange
+        final String PASSWORD = "thisIsABadPassword";
+
+        final List<String> INVALID_SALTS = Arrays.asList("argon2", "$3a$11$", "x", "$2a$10$");
+        final String LENGTH_MESSAGE = "The raw salt must be greater than or equal to 8 bytes";
+
+        EncryptionMethod encryptionMethod = EncryptionMethod.AES_CBC;
+
+        // Act
+        for (final String salt : INVALID_SALTS) {
+            IllegalArgumentException iae = assertThrows(IllegalArgumentException.class,
+                    () -> cipherProvider.getCipher(encryptionMethod, PASSWORD, salt.getBytes(), DEFAULT_KEY_LENGTH, true));
+
+            // Assert
+            assertTrue(iae.getMessage().contains(LENGTH_MESSAGE));
+        }
+    }
+
+    @Test
+    void testGetCipherShouldHandleUnformattedSalts() throws Exception {
+        // Arrange
+        final String PASSWORD = "thisIsABadPassword";
```

Comment:
It would be helpful to refactor this to a shared static value instead of repeating it in all methods.

Reply:
Moved to a static values